packages/devtools-launchpad/src/menu.js

```diff
@@ -0,0 +1,88 @@
+const { Menu, MenuItem } = require("devtools-sham-modules");
+const { isFirefoxPanel } = require("devtools-config");
+
+function createPopup(doc) {
+  let popup = doc.createElement("menupopup");
+  popup.className = "landing-popup";
+  if (popup.openPopupAtScreen) {
+    return popup;
+  }
+
+  function preventDefault(e) {
+    e.preventDefault();
+    e.returnValue = false;
+  }
+
+  let mask = document.querySelector("#contextmenu-mask");
+  if (!mask) {
+    mask = doc.createElement("div");
+    mask.id = "contextmenu-mask";
+    document.body.appendChild(mask);
+  }
+
+  mask.onclick = () => popup.hidePopup();
+
+  popup.openPopupAtScreen = function(clientX, clientY) {
+    this.style.setProperty("left", `${clientX}px`);
+    this.style.setProperty("top", `${clientY}px`);
+    mask = document.querySelector("#contextmenu-mask");
+    window.onwheel = preventDefault;
+    mask.classList.add("show");
+    this.dispatchEvent(new Event("popupshown"));
+    this.popupshown;
+  };
+
+  popup.hidePopup = function() {
+    this.remove();
+    mask = document.querySelector("#contextmenu-mask");
+    mask.classList.remove("show");
+    window.onwheel = null;
+  };
+
+  return popup;
+}
+
+if (!isFirefoxPanel()) {
+  Menu.prototype.createPopup = createPopup;
+}
+
+function onShown(menu, popup) {
+  popup.childNodes.forEach((menuitem, index) => {
+    const item = menu.items[index];
+
+    if (!item.disabled) {
+      menuitem.onclick = () => {
+        item.click();
+        popup.hidePopup();
+      };
+    }
+  });
+}
+
+function showMenu(e, items) {
+  if (items.length === 0) {
+    return;
+  }
+
+  const menu = new Menu();
+  items.forEach(item => menu.append(new MenuItem(item)));
+
+  if (isFirefoxPanel()) {
+    return menu.popup(e.screenX, e.screenY, { doc: window.parent.document });
+  }
+
+  menu.on("open", (_, popup) => onShown(menu, popup));
+  menu.popup(e.clientX, e.clientY, { doc: document });
+}
+
+function buildMenu(items) {
+  return items.map(itm => {
+    const hide = typeof itm.hidden === "function" ? itm.hidden() : itm.hidden;
+    return hide ? null : itm.item;
+  }).filter(itm => itm !== null);
+}
+
+module.exports = {
+  showMenu,
+  buildMenu
+};
```

Comment:
nice!

Lets also export these methods in `src/index` so that we can kill the debugger implementation :)

https://github.com/devtools-html/devtools-core/blob/master/packages/devtools-launchpad/src/index.js#L131-L135